docs/source/tutorials/debugging.rst

```diff
@@ -0,0 +1,66 @@
+Debugging DepthAI pipeline
+##########################
+
+Currently, tools for debugging the DepthAI pipeline are limited. We plan on creating a software that would track all messages and queues,
+which would allow users to debug a "frozen" pipeline much easier, which is usually caused by a filled up :ref:`blocking queue <Blocking behaviour>`.
+
+DepthAI debugging level
+=======================
+
+You can enable debugging by changing the debugging level. It's set to :code:`error` by default.
+
+================  =======
+Level             Logging
+================  =======
+:code:`critical`  Only a critical error that stops/crashes the program.
+:code:`error`     Errors will not stop the program, but won't complete the action. Examples:
+                    - When :ref:`ImageManip` cropping ROI was out of bounds, error will get printed and the cropping won't take place
+
+                    - When :ref:`NeuralNetwork` gets a frame whose shape (width/heigth/channel) isn't that of the :code:`.blob`
+:code:`warn`      Warnings (add example here)
```

Comment:
Warnings are printed in cases where users action could improve certain behavior or fix them if it was intended.